tests/results/result_actions/ApplyPatchActionTest.py

```diff
@@ -91,6 +91,51 @@ def test_apply_orig_option(self):
 
             self.assertEqual(file_dict, expected_file_dict)
 
+    def test_apply_rename(self):
+        uut = ApplyPatchAction()
+        with make_temp() as f_a:
+            file_dict = {
+                f_a: ["1\n", "2\n", "3\n"]
+                }
+            expected_file_dict = {
+                f_a: ["1\n", "2_changed\n", "3_changed\n"]
+                }
+            file_diff_dict = {}
+            diff = Diff(file_dict[f_a], rename=f_a+".renamed")
+            diff.change_line(3, "3\n", "3_changed\n")
+            uut.apply(Result("origin", "msg", diffs={f_a: diff}),
+                      file_dict,
+                      file_diff_dict)
+            self.assertTrue(isfile(f_a+".orig"))
+            self.assertTrue(isfile(f_a+".renamed"))
+
+            diff = Diff(file_dict[f_a])
+            diff.change_line(2, "2\n", "2_changed\n")
+            uut.apply(Result("origin", "msg", diffs={f_a: diff}),
+                      file_dict,
+                      file_diff_dict)
+            self.assertTrue(isfile(f_a+".renamed.orig"))
+
+            for filename in file_diff_dict:
+                file_dict[filename] = file_diff_dict[filename].modified
+
+            self.assertEqual(file_dict, expected_file_dict)
+
+    def test_apply_delete(self):
+        uut = ApplyPatchAction()
+        with make_temp() as f_a:
+            file_dict = {
+                f_a: ["1\n", "2\n", "3\n"]
+                }
+            file_diff_dict = {}
+            diff = Diff(file_dict[f_a], delete=True)
+            uut.apply(Result("origin", "msg", diffs={f_a: diff}),
+                      file_dict,
+                      file_diff_dict)
+
+            self.assertFalse(isfile(f_a))
+            open(f_a, 'w')
```

Comment:
try applying another patch that has the delete flag also set, there should be no conflict and nothing happening because all is well (but I think your current routine would attemt to delete it again)